schema/schema.py

```diff
@@ -109,18 +108,65 @@ def generate_schema() -> dict:
     schema["type"] = "object"
     schema["properties"] = {
         # Version number required for samconfig files to be valid
-        "version": {"type": "number"}
+        "version": {"title": "Config version", "type": "number", "default": 0.1}
     }
     schema["required"] = ["version"]
     schema["additionalProperties"] = False
     # Iterate through packages for command and parameter information
     for package_name in _SAM_CLI_COMMAND_PACKAGES:
         new_command = retrieve_command_structure(package_name)
         commands.update(new_command)
-        for param_list in new_command.values():
+    for cmd_properties in commands.values():
+        for param_list in cmd_properties["parameters"].values():
             command_params = [param for param in param_list]
         params.update(command_params)
-    # TODO: Generate schema for each of the commands
+    # Generate schema for each of the commands
+    schema["patternProperties"] = {SchemaKeys.ENVIRONMENT_REGEX.value: {"title": "Environment", "properties": {}}}
+    COMMANDS_TO_EXCLUDE = [  # TEMPORARY: for use only while generating piece-by-piece
+        "deploy",
+        "build",
+        "local",
+        "validate",
+        "package",
+        "init",
+        "delete",
+        "bootstrap",
+        "list",
+        "traces",
+        "sync",
+        "publish",
+        "pipeline",
+        "logs",
+        "remote",
+    ]
+    for cmd_name, cmd_items in commands.items():
+        full_cmd_name = cmd_name.split("_")
+        formatted_cmd_name = " ".join(full_cmd_name)
+        exclude_params = full_cmd_name[0] in COMMANDS_TO_EXCLUDE
+        formatted_params_list = (
+            "* "
+            + "\n* ".join([f"{param['title']}:\n{param['description']}" for param in cmd_items["parameters"].values()])
+            if not exclude_params
+            else ""
+        )
+        param_description = f"Available parameters for the {formatted_cmd_name} command:\n{formatted_params_list}"
+        schema["patternProperties"][SchemaKeys.ENVIRONMENT_REGEX.value]["properties"].update(
+            {
+                cmd_name: {
+                    "title": f"{' '.join(full_cmd_name).title()} command",
+                    "description": cmd_items["description"] or "",
```

Comment:
I've noticed that some of the descriptions in the skeleton schema have tabs in them, will this affect anything in how the information is displayed in the various code editors/IDEs? (this isn't the location where the description is set, but wanted to call it out here)

Also `.get()` to access this (this change is dependent on the outcome of the comment at the top of this file) :)

Reply:
In my testing using VSCode, it doesn't seem to be an issue for readability. I thought it was odd that there's inconsistency among the descriptions, with some starting with newlines or being indented more than others, but at least with VSCode it's still legible. 
I could look into potentially reformatting some of the help text to be more consistent with everything, but I'm worried that may have impact across the CLI itself (i.e. things are formatted the way they are for readability in the CLI, and fixing for the description could decrease readability there). Alternatively, we could write a cleaner function to do things like prevent more than two spaces between words, and prevent more than one or two newlines.

Reply:
Just to pull in an example from the build skeleton: https://github.com/aws/aws-sam-cli/pull/5484/files#diff-8ad631c7b37df828ce338f907037667605bd5eaad2e91c772c073936cd90a5f6R52

If its pulling the descriptions right from the commands, then the formatting is likely there to make it easier to read when running something like `sam build --help`. We could also consider just trimming the start and ends of the descriptions make sure that they don't get parsed by other extensions or editors outside of VSCode.

Reply:
Trimming off newlines and extra spaces from the ends would probably be best in that case. If there are other weird formatting issues we come across afterwards, we can address them as they come up, such as with the specific command schemas.